client/network-gossip/src/state_machine.rs

```diff
@@ -429,6 +454,24 @@ impl<B: BlockT> ConsensusGossip<B> {
 	}
 }
 
+struct Metrics {
+	messages: Gauge<U64>,
+}
+
+impl Metrics {
+	fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			messages: register(
+				Gauge::new(
```

Comment:
In order to notice peaks in between Prometheus scrape intervals, I would suggest using two counters. One increased when a new message is registered, one increased each time a message expires. The current number of messages can be calculated by subtracting the latter metric from the former.

What do you think? Does the above reasoning make sense?